Tools/automation.ps1

```diff
@@ -0,0 +1,791 @@
+        {
+            ## Modules have been identified as missing
+            $ErrorMessage = "`n`nMissing required PowerShell modules`n"
+            $ErrorMessage += "    Run the following command to install the missing modules: Import-Module Az`n"
+            
+            Write-Host $ErrorMessage -ForegroundColor Yellow
+            Throw "Unable to run script, missing required PowerShell modules..."
+        }
+
+        ## Create Folders for the Parameter and Template folder paths
+        $Result = New-Item -ItemType Directory -Path $ParameterFolderPath -ErrorAction SilentlyContinue -InformationAction SilentlyContinue
+        If($Result)
+            { Write-Host "Created Directory to contain the ARM Parameter files ($($Result.FullName))." }
+
+        #### Connect to Azure ####
+        $Result = Connect-Azure -AzSubscriptionName $AzSubscriptionName
+
+        ## If the connection to azure attempt fails.. then exit.
+        If(!$($Result))
+            { Throw "Failed to connect to Azure" }
+
+        #### Create Resource Group ####
+        New-AzureResourceGroup -AzResourceGroupName $AzResourceGroup
+
+        #### Creates the ARM Parameter files ####
+        $ARMObjects = New-ARMParameterObject -ParameterFolderPath $ParameterFolderPath -TemplateFolderPath $TemplateFolderPath -Index $Index -ResourcePrefix $ResourcePrefix -AzLocation $AzLocation
+
+        #### Verifies ARM Parameters are correct ####
+        $Result = Test-ARMTemplate -ARMObjects $ARMObjects
+
+        ## If the attempt fails.. then exit.
+        If($($Result))
+            { Throw "ARM Template and Parameter testing failed" }
+
+        #### Creates Azure Objects with ARM Templates and Parameters ####
+        New-ARMObjects -ARMObjects $ARMObjects -ArchiveFunctionZip "$WorkingDirectory\CompiledFunctions.zip" -AzResourceGroup $AzResourceGroup
+
+        #Creates a spacing between the last step and the next
+        Write-Host "`n"
+
+        ## Shows how to connect local Windows Package Manager Client to newly created private repository
+        IF($ShowConnectionInstructions)
+        {
+            $AzSubscriptionName = $(Get-AzContext).Subscription.Name
+            $jsonFunction       = Get-Content -Path $($ARMObjects.Where({$_.ObjectType -eq "Function"}).ParameterPath) | ConvertFrom-Json
+            $AzFunctionName     = $jsonFunction.Parameters.FunctionName.Value
+            $AzFunctionURL      = $(Get-AzFunctionApp -Name $AzFunctionName -ResourceGroupName $AzResourceGroup).DefaultHostName
+
+            ## Post script Run Informational:
+            #### Instructions on how to add the repository to your Windows Package Manager Client
+            Write-Host "Use the following command to register the new private repository with your Windows Package Manager Client:" -ForegroundColor Yellow
+            Write-Host "  winget source add -n ""PrivateRepo"" -a ""https://$AzFunctionURL/api/"" -t ""Microsoft.Rest""" -ForegroundColor Yellow
```

Comment:
Color check, is yellow the correct color to use?

Reply:
Not sure I understand...?